lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1475,6 +1495,17 @@ public function get_attribute( $name ) {
 
 		$attribute = $this->attributes[ $comparable ];
 
+		/*
+		 * This flag distinguishes an attribute with no value
+		 * from an attribute with an empty string value. For
```

Comment:
I must have missed the discussion – we could distinguish between NULL values, false values, and empty strings. Is the `is_true` flag a poka yoke/a developer experience affordance to make it more convenient to reason about? Or is there more to it?

Reply:
There's more to it and I'm pretty sure it'd come back to you if you looked, as it was the case for me.

First of all, there's no `false` attribute value in HTML. Only `true`. An attribute is either a boolean `true` or it's absent. The DOM API doesn't even expose this. There is no way to differentiate an attribute with an empty string value and a true value, but `setAttribute( 'wp-test', '' )` omits the string value when creating the HTML representation of the node. The DOM API is more matter-of-fact/doesn't matter about it than we are, but we're attempting to represent syntactical equivalence rather than semantic.

https://user-images.githubusercontent.com/5431237/214435276-745da71e-d8fb-481d-860b-3d584a89b3f7.mov

In other words this exists to differentiate cases like `enabled` and `enabled=enabled` and `enabled=""`